commit/merkleroot/observation.go

```diff
@@ -202,6 +205,55 @@ func (o ObserverImpl) ObserveOffRampNextSeqNums(ctx context.Context) []plugintyp
 	return result
 }
 
+// ObserveLatestOnRampSeqNums observes the latest onRamp sequence numbers for each configured source chain.
+func (o ObserverImpl) ObserveLatestOnRampSeqNums(
+	ctx context.Context, destChain cciptypes.ChainSelector) []plugintypes.SeqNumChain {
+
+	allSourceChains, err := o.chainSupport.KnownSourceChainsSlice()
+	if err != nil {
+		o.lggr.Warnw("call to KnownSourceChainsSlice failed", "err", err)
+		return nil
+	}
+
+	supportedChains, err := o.chainSupport.SupportedChains(o.nodeID)
+	if err != nil {
+		o.lggr.Warnw("call to KnownSourceChainsSlice failed", "err", err)
+		return nil
+	}
+
+	sourceChains := mapset.NewSet(allSourceChains...).Intersect(supportedChains).ToSlice()
+	sort.Slice(sourceChains, func(i, j int) bool { return sourceChains[i] < sourceChains[j] })
+
+	latestOnRampSeqNums := make([]plugintypes.SeqNumChain, len(sourceChains))
+	eg := &errgroup.Group{}
+
+	for i, sourceChain := range sourceChains {
+		i, sourceChain := i, sourceChain
+		eg.Go(func() error {
+			nextOnRampSeqNum, err := o.ccipReader.GetExpectedNextSequenceNumber(ctx, sourceChain, destChain)
+			if err != nil {
+				return fmt.Errorf("failed to get expected next sequence number for source chain %d: %w", sourceChain, err)
+			}
+			if nextOnRampSeqNum == 0 {
+				return fmt.Errorf("expected next sequence number for source chain %d is 0", sourceChain)
+			}
+
+			latestOnRampSeqNums[i] = plugintypes.SeqNumChain{
+				ChainSel: sourceChain,
+				SeqNum:   nextOnRampSeqNum - 1, // Latest is the next one minus one.
```

Comment:
`latestOnRampSeqNum` would be set to 0 for a new chain. Is that OK?

Reply:
yes, should be good.

Reply:
yeah, in that case there won't be any range reported, since (onRampMax < offRampNext).